bundles/org.openhab.binding.exec/src/main/java/org/openhab/binding/exec/internal/handler/ExecHandler.java

```diff
@@ -160,21 +172,61 @@ public void run() {
                         commandLine = String.format(commandLine, Calendar.getInstance().getTime());
                     }
                 } catch (IllegalFormatException e) {
-                    logger.error(
+                    logger.warn(
                             "An exception occurred while formatting the command line with the current time and input values : '{}'",
                             e.getMessage());
                     updateState(RUN, OnOffType.OFF);
                     return;
                 }
 
-                logger.trace("The command to be executed will be '{}'", commandLine);
+                String[] cmdArray;
+                String[] shell;
+                if (commandLine.contains(CMD_LINE_DELIMITER)) {
+                    logger.debug("Splitting by '{}'", CMD_LINE_DELIMITER);
+                    try {
+                        cmdArray = commandLine.split(CMD_LINE_DELIMITER);
+                    } catch (PatternSyntaxException e) {
+                        logger.warn("An exception occurred while splitting '{}' : '{}'",
+                                commandLine, e.getMessage());
+                        updateState(RUN, OnOffType.OFF);
+                        updateState(OUTPUT, new StringType(e.getMessage()));
+                        return;
+                    }
+                } else {
+                    // Invoke shell with 'c' option and pass string
+                    logger.debug("Passing to shell for parsing command.");
+                    if (getOperatingSystemType() == OS.WINDOWS) {
```

Comment:
I have a question here: Why don't you get the operating system type once and then use the local variable in a switch-statement. Are your sure that everything other than `OS.WINDOWS` and `OS.UNKNOWN` is a supported *nix? And please add the unsupported OS-name to the warn-message `logger.warn("OS '{}' not supported, please split commands manually!", getOperatingSystemName());`).

Edit: I found it below. But the if-else is not very intuitive here. Please use a switch-statement.